test/test_common/environment.cc

```diff
@@ -65,12 +68,17 @@ bool TestEnvironment::shouldRunTestForIpVersion(const Network::Address::IpVersio
 
 std::vector<Network::Address::IpVersion> TestEnvironment::getIpVersionsForTest() {
   std::vector<Network::Address::IpVersion> parameters;
-  if (TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v4)) {
-    parameters.push_back(Network::Address::IpVersion::v4);
-  }
-
-  if (TestEnvironment::shouldRunTestForIpVersion(Network::Address::IpVersion::v6)) {
-    parameters.push_back(Network::Address::IpVersion::v6);
+  for (auto version : {Network::Address::IpVersion::v4, Network::Address::IpVersion::v6}) {
+    if (TestEnvironment::shouldRunTestForIpVersion(version)) {
+      parameters.push_back(version);
+      if (!Network::Test::supportsIpVersion(version)) {
+        Logger::Registry::getLog(Logger::Id::testing)
+            .warn(
```

Comment:
I'm thinking there are cases where this check is either too strong or too weak. So currently it only provides information to the user about why the tests may have failed. There may be some tests that only create sockets but don't bind to them. This seems to succeed even in IP version only environments. Also, I'm thinking there may be tests that require more than the ability to bind to a loopback IP version address.